api-tests/api.test.js

```diff
@@ -89,3 +102,93 @@ test('/events endpoint', async () => {
     ])
   })
 })
+
+test('/events/{eventKey}/info endpoint', async () => {
+  const d = await fetch(addr + '/events/2018flor/info').then(d => d.json())
+  const info = d.data
+  expect(info.key).toEqual('2018flor')
+  expect(info.name).toBeA(String)
+  expect(info.startDate).toBeADateString()
+  expect(info.endDate).toBeADateString()
+  expect(info.location).toBeA(Object)
+  expect(info.location.name).toBeA(String)
+  expect(info.location.lat).toBeA(Number)
+  expect(info.location.lon).toBeA(Number)
+  expect(info.key).toBeA(String)
+  expect(info.district).toBeUndefinedOr(String)
+  expect(info.week).toBeUndefinedOr(Number)
+  expect(info.webcasts).toEqual(expect.any(Array))
+  info.webcasts.forEach(webcast => {
+    expect(webcast).toEqual({
+      url: expect.any(String),
+      type: expect.stringMatching(youtubeOrTwitch),
+    })
+  })
+  expect(Object.keys(info)).toBeASubsetOf([
+    'key',
+    'name',
+    'week',
+    'startDate',
+    'endDate',
+    'location',
+    'district',
+    'webcasts',
+  ])
+})
+
+test('/events/{eventKey}/matches endpoint', async () => {
+  const d = await fetch(addr + '/events/2018flor/matches').then(d => d.json())
+  expect(d).toEqual({ data: expect.any(Array) })
+  expect(d.data.length).toBeGreaterThan(1)
+  d.data.forEach(match => {
+    expect(match.key).toBeA(String)
+    expect(match.time).toBeADateString()
+    expect(match.redScore).toBeUndefinedOr(Number)
+    expect(match.blueScore).toBeUndefinedOr(Number)
+    expect(match.redAlliance).toEqual(expect.any(Array))
+    match.redAlliance.forEach(team => {
+      expect(team).toBeATeamKey()
+    })
+    expect(match.blueAlliance).toEqual(expect.any(Array))
+    match.blueAlliance.forEach(team => {
+      expect(team).toBeATeamKey()
+    })
+    expect(Object.keys(match)).toBeASubsetOf([
+      'key',
+      'time',
+      'redAlliance',
+      'blueAlliance',
+      'redScore',
+      'blueScore',
+    ])
+  })
+})
+
+test('/events/{eventKey}/matches/{matchKey}/info endpoint', async () => {
+  const d = await fetch(
+    addr + '/events/2018flor/matches/2018flor_qm28/info',
+  ).then(d => d.json())
+  const info = d.data
+  expect(info.key).toEqual('2018flor_qm28')
+  expect(info.time).toBeADateString()
+  expect(info.redScore).toBeUndefinedOr(Number)
+  expect(info.blueScore).toBeUndefinedOr(Number)
+  expect(info.redAlliance).toEqual(expect.any(Array))
+  expect(info.redAlliance.length).toEqual(3)
+  info.redAlliance.forEach(team => {
+    expect(team).toBeATeamKey()
+  })
+  expect(info.blueAlliance).toEqual(expect.any(Array))
+  expect(info.blueAlliance.length).toEqual(3)
```

Comment:
https://jestjs.io/docs/en/expect#tohavelengthnumber

This will give you a nicer error message when it fails